### ISSUE TITLE
Roadmap Plans: Route from openff-interchange to replacement of InterMol?

### ISSUE THREAD

**Issue by mrshirts** [openforcefield/openff-interchange#284]
Just wanted to document and discuss the routes for openff-interchange replacing the functionality of InterMol. Note that there is no guarantee that all of these things will supported at any particular timeline, but based on the timeline for openff-interchange, many of these will be supported at some time.

- Write from internal representation to:
   - GROMACS: done?
   - CHARMM
   - AMBER
   - DESMOND
   - OpenMM: done
   - LAMMPS: done?

- READ into internal representation from:
   - GROMACS
   - CHARMM
   - AMBER
   - DESMOND
   - LAMMPS
   - OpenMM
  
  There is currently no specific timeline for writing importers; the expectation is that they will be handled by developers of those programs OR added as needed when needed. 

- Energy validation of "in" and "out" file formats
   - Basically, Interchange will always make sure that if there is an output writer, the energies will be evaluated "natively" by the program of conversion into.  
   - Presumably it will be possible to adapt this functionality to evaluate energies of input files when importers are developed.

- Energy functions that may still need to be implemented.
    - Free energy notations (note, **not** in InterMol). 
    - Virtual Sites
    - CMAP
    - SETTLES/constraints

**Comment by mattwthompson** [comment]
In general I'm not sure what the goalpost for being "done" should be; getting simple systems to work can take as little as a few minutes (if using inelegant patterns) and on the other end, there is basically no ceiling for the scope of things that could be done. Other difficulties with discretizing the objective are that correctness is difficult to define and the data sources are heterogeneous and (with few exceptions) not specified.


Pretty much everything in this direction is blocked by importer(s) being written; since there's no cohesive internal representation in InterMol, I don't think there's a way to test conversions without `Interchange.from_engineX`. On the upside, however, once one importer X is written, the pathways from X to all other engines can be tested immediately (or at least that's the design). A middle ground would be the non-public ParmEd converters, but that's blocked by the topology refactor in the toolkit being completely done, which is not likely to be reached soon.

The only engine missing from these lists is OpenMM, in which the "to" support is pretty good and the "from" support is spotty, since it's not as common a pathway (if/until we think about how to interface with `openmmforcefields` ...)

I  personally have no experience with DESMOND and have not reviewed its license, so until we learn of people wishing for support and/or external contributors willing to work with the project, I'd like to leave it on the backlog.


> Free energy notations

Could you expand one this? I don't quite understand what is meant here

Some other mildly exotic functions that are, are expected to be, or may be included:
* Bond order-based interpolation of parameters (SMIRNOFF only, bonds/torsions only as of now)
* Non-standard non-bonded functional forms (i.e. anything but 12-6 LJ)
* Non-harmonic bond or angle terms
* Urey-Bradleyterms
* 1-3 or 1-5 intramolecular scalings to anything but 0 or 1, respectively

**Comment by mrshirts** [comment]
> Free energy notations

That was vague. "Free energy notations" would be data that indicates a topology file actually contains two molecules that are alchemically transformed.  There's a couple of different ways to do it (hence hard), either indicating two entirely different molecules and what atoms they share, or flagging interaction sites or bonds that change, and listing two sets of parameters for each of them.  It's complicated - InterMol never did it -- but it is a type of file that might be encountered sometime, since one may have a tool in one program that plans out a series of free energy calculations, and one might want to convert that to some other output.

**Comment by mrshirts** [comment]
That list of interaction terms looks fairly comprehensive.  I don't THINK 1-3's are likely to occur, since that is the angle term, but maybe I am wrong. I haven't seen 1-5's implemented, and I think it might be kind of complicated to understand (since 1-5's usually have a couple of degrees of internal freedom between them, not just a dihedral DOF) but maybe I'm wrong there as well.

**Comment by mrshirts** [comment]
> In general I'm not sure what the goalpost for being "done" should be;

I don't know there is a goalpost, since interchange OpenFF  goals for near term are only partly aligned with any broader interchange effort.  But keeping track of what is on roadmap and which isn't is useful so anyone interested in broader usage can see what remains.

**Comment by mrshirts** [comment]
> Pretty much everything in this direction is blocked by importer(s) being written; since there's no cohesive internal representation in InterMol,

Yes, I think the near term goal enabling this is to make sure the API for getting anything into the interchange format is working and well-documented.

**Comment by mattwthompson** [comment]
> Free energy-focused topologies

In general this sounds like a set of features that could be supported, but until a specification for this model is in place it would be difficult to make any progress.

**Comment by mattwthompson** [comment]
I'm going to close this for now only because of how outdated it is

The state of support is tracked here: https://docs.openforcefield.org/projects/interchange/en/stable/using/status.html
